### PR TITLE
SLS-51 Write disk image for update restore eviction

### PR DESCRIPTION
**Pull request by quchenhao**
We currently skip write the disk image to disk if it's an update restore eviction because it will be written again by the checkpoint later anyway. However, it is not correct if we have delta. We need to ensure the delta is built on a disk image that has been written to disk. Change to always write the disk image even for update restore eviction.

### CODE REVIEW

File: src/btree/bt_split.c [hunk 1]
```diff
@@ -2318,6 +2320,19 @@ __wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi)
...
+    if (multi->addr.block_cookie != NULL) {
+        WT_ERR(__wt_calloc_one(session, &addr));
+        WT_TIME_AGGREGATE_COPY(&addr->ta, &multi->addr.ta);
+        WT_ERR(__wt_memdup(
+          session, multi->addr.block_cookie, multi->addr.block_cookie_size, &addr->block_cookie));
+        addr->block_cookie_size = multi->addr.block_cookie_size;
+        addr->type = multi->addr.type;
+        __wt_ref_addr_free(session, ref);
+        ref->addr = addr;
+    }
```
Comment (korteland) [inline src/btree/bt_split.c:2334]: Higher-level design question: does giving the ref an address mean it's going to get written out later? Trying to piece the story together from the ticket + PR.

File: src/reconcile/rec_write.c [hunk 2]
```diff
@@ -2414,8 +2414,11 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         if (r->page->type != WT_PAGE_ROW_LEAF && chunk->entries == 0)
             return (__wt_set_return(session, EBUSY));
 
-        /* If we need to restore the page to memory, copy the disk image. */
-        if (multi->supd_restore)
+        /*
+         * If we need to restore the page to memory, copy the disk image if it is not disaggregated
+         * or it is empty.
```
Comment (korteland) [inline src/reconcile/rec_write.c:2419]: I could use a slightly expanded comment explaining why these are exceptions.

File: src/btree/bt_split.c [hunk 3]
```diff
@@ -2318,6 +2320,19 @@ __wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi)
     __wt_page_modify_clear(session, page);
     if (!F_ISSET(S2C(session)->cache, WT_CACHE_EVICT_SCRUB) || multi->supd_restore)
         F_SET_ATOMIC_16(page, WT_PAGE_EVICT_NO_PROGRESS);
+
+    /* If there's an address, copy it. */
+    if (multi->addr.block_cookie != NULL) {
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply src/btree/bt_split.c:2325]: The ref links to a __wt_addr struct not a __wt_addr_copy. In the addr copy, we have the page deleted infrastructure. In __wt_addr we don't have that. Also __wt_addr_copy is used when there is concurrent concern. Here we are thread safe as eviction has exclusive access.

File: src/btree/bt_split.c (hunk 1 above)
Comment (quchenhao) [reply src/btree/bt_split.c:2334]: Yes, checkpoint may write that address in the internal page. We also need to be able to free the address when a new reconciliation happen.

File: src/reconcile/rec_write.c (hunk 2 above)
Comment (quchenhao) [reply src/reconcile/rec_write.c]: fixed

File: src/btree/bt_split.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (korteland) [reply src/btree/bt_split.c:2325]: Ah, ok. Thanks.

**Review by korteland — APPROVED** [review]
LGTM!

**Review by pmacko86 — APPROVED** [review]
LGTM. I don't understand the reconciliation code well, so this was just a cursory review.

**Pull request merged**